### PR TITLE
feat: added unique constraint & logic to mutation

### LINKED ISSUE #425: Add uniques to WorkHour (day, userId, taskId) in db
- update prisma schema
- make migration
- `workHourCreate` should add the new duration to existing entries.
- update tests

Story Points: 3

### PR DESCRIPTION
**Pull request by moncapitaine**
fix #425

### CODE REVIEW

File: app/backend/prisma/migrations/20220902151808_workhours_unique_day/migration.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+  Warnings:
+
+  - A unique constraint covering the columns `[date,userId,taskId]` on the table `WorkHour` will be added. If there are existing duplicate values, this will fail.
```
Comment (PascalHelbig) [inline app/backend/prisma/migrations/20220902151808_workhours_unique_day/migration.sql:4]: Let's see if this will work :)

**Review by PascalHelbig — APPROVED** [review]
Looks good

File: app/backend/graphql/workHour/mutations/workHourCreateMutationField.ts [hunk 2]
```diff
@@ -31,18 +31,26 @@ export const workHourCreateMutationField = mutationField('workHourCreate', {
...
+    const oldRecord = await context.prisma.workHour.findUnique({ where: { date_userId_taskId: workHourKey } })
+
+    if (oldRecord) {
+      return context.prisma.workHour.update({
+        where: { date_userId_taskId: workHourKey },
+        data: { duration: oldRecord.duration + arguments_.data.duration },
+      })
+    }
+
+    return context.prisma.workHour.create({ data: { ...workHourKey, duration: arguments_.data.duration } })
```
Comment (PascalHelbig) [inline app/backend/graphql/workHour/mutations/workHourCreateMutationField.ts:54]: How about using the `.upsert()` method here?

In Prisma there is increment option for updating: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#increment-all-view-and-likes-fields-of-all-post-records-by-1

**Review by PascalHelbig — CHANGES_REQUESTED** [review]

File: app/backend/prisma/migrations/20220902151808_workhours_unique_day/migration.sql (hunk 1 above)
Comment (moncapitaine) [reply app/backend/prisma/migrations/20220902151808_workhours_unique_day/migration.sql]: I did the work now to transform the data to fit into the unique constraint now, please review. Just did manual tests locally and it works on my machine

File: app/backend/graphql/workHour/mutations/workHourCreateMutationField.ts (hunk 2 above)
Comment (moncapitaine) [reply app/backend/graphql/workHour/mutations/workHourCreateMutationField.ts]: done

**Review by PascalHelbig — APPROVED** [review]

**Pull request merged**